src/logic/GameLogic.cs

```diff
@@ -60,15 +61,39 @@ public class GameLogic {
 		/// <summary> Handles all on-startup tasks, instantiation of objects, or other similar run-once tasks. </summary>
 		public virtual void Initialize() {
 			ItemDefinition.LoadDefinitions();
-			Level = new Level();
+			Player = new Player(new Vector2(0.0f, 0.0f));
+			Level = LevelGenerator.TryGenerateLevel();
+			Level.Player = Player;
+			Renderer.EventQueue.Enqueue("LevelRegenerated"); //Signal to renderer to regenerate map scene
 			Server = new Server(Level);
 			Sounds.Init();
 		}
 
 		/// <summary> Primary gameplay loop. Make all your calls and modifications to State, not StateBuffer!</summary>
-		public virtual void Update() {
+		public virtual void Update(double deltaTime) {
+			//Regenerate level if signalled
+			if (Level.NeedsRegen) {
+				uint score = Level.Score;
+				Level = LevelGenerator.TryGenerateLevel(Level.Depth);
+				Level.Score = score;
+				Level.Player = Player;
+				Renderer.EventQueue.Enqueue("LevelRegenerated"); //Signal to renderer to regenerate map scene
+			}
+
+			//Player died, reset level
+			if (Player.Health <= 0) {
+				Level.CurrentRoom.OnExit(Level, Level.StartRoom); //Call on exit so things like ambient music are stopped
+				Sounds.PlaySound("assets/sounds/PlayerDeath0.wav"); //Play death sound
+				//Recreate player and generate new level
+				Player = new Player(new Vector2(0.0f, 0.0f));
+				Level = LevelGenerator.TryGenerateLevel();
+				Level.Player = Player;
+				Console.WriteLine($"Player died with a score of {Level.Score}!");
+				Renderer.EventQueue.Enqueue("LevelRegenerated"); //Signal to renderer to regenerate map scene
```

Comment:
Later, this should be changed to a "PlayerDeath" so can I can implement a death screen. For now, good.